packages/firestore/src/remote/remote_store.ts

```diff
@@ -155,25 +155,26 @@ export class RemoteStore {
 
   /**
    * Updates our OnlineState to the new state, updating local state
-   * and notifying the onlineStateHandler as appropriate.
+   * and notifying the onlineStateHandler as appropriate. Idempotent.
    */
-  private updateOnlineState(newState: OnlineState) {
+  private updateOnlineState(newState: OnlineState): void {
     if (newState === OnlineState.Healthy) {
       // We've connected to watch at least once. Don't warn the developer about
       // being offline going forward.
       this.shouldWarnOffline = false;
-    } else if (newState === OnlineState.Unknown) {
-      // The state is set to unknown when a healthy stream is closed (e.g. due to
-      // a token timeout) or when we have no active listens and therefore there's
-      // no need to start the stream. Assuming there is (possibly in the future)
-      // an active listen, then we will eventually move to state Online or Failed,
-      // but we always want to make at least ONLINE_ATTEMPTS_BEFORE_FAILURE
-      // attempts before failing, so we reset the count here.
-      this.watchStreamFailures = 0;
     }
 
     // Update and broadcast the new state.
```

Comment:
Can you move this comment closer to the actual assignment / broadcast?

Reply:
Ended up dropping this comment. I think it was just describing what the two lines were doing, and not why and therefore is not super helpful.